File: schemaorg/answer.go
package schemaorg

import "encoding/json"

// Answer see : https://schema.org/Answer
type Answer struct {

Comment

typeContext

// DownvoteCount see : https://schema.org/downvoteCount
// The number of downvotes this question, answer or comment has received from the community.
DownvoteCount int `json:"downvoteCount"`

// ParentItem see : https://schema.org/parentItem
// The parent of a question, answer or item in general.
ParentItem *Question `json:"parentItem"`

// UpvoteCount see : https://schema.org/upvoteCount
// The number of upvotes this question, answer or comment has received from the community.
UpvoteCount int `json:"upvoteCount"`

}

func (v *Answer) MarshalJSON() ([]byte, error) {
	v.C = "http://schema.org"
	v.T = "Answer"

	return json.Marshal(v)
}

File: schemaorgphp/classes/answer.php
<?php

class Answer extends Comment implements JsonSerializable {

	public static $context = 'http://schema.org';
	public static $type = 'Answer';
	
	/**
	 * The number of downvotes this question, answer or comment has received from the community.
	 * see : https://schema.org/downvoteCount
	 * @var integer|integer[]
	 */
	public var $downvote_count;
	
	/**
	 * The parent of a question, answer or item in general.
	 * see : https://schema.org/parentItem
	 * @var \Question|\Question[]
	 */
	public var $parent_item;
	
	/**
	 * The number of upvotes this question, answer or comment has received from the community.
	 * see : https://schema.org/upvoteCount
	 * @var integer|integer[]
	 */
	public var $upvote_count;
	
	public function jsonSerialize() {
		$out = array(
			'@context' => 'http://schema.org',
			'@type' => 'Answer'
		);
		
		$serialized = so_json_serialize( $this->downvote_count );
		if ( ! empty( $serialized ) ) {
			$out['downvoteCount'] = $serialized;
		}
		
		$serialized = so_json_serialize( $this->parent_item );
		if ( ! empty( $serialized ) ) {
			$out['parentItem'] = $serialized;
		}
		
		$serialized = so_json_serialize( $this->upvote_count );
		if ( ! empty( $serialized ) ) {
			$out['upvoteCount'] = $serialized;
		}
		
		return $out;
	}
}
